frontend/packages/client/src/components/Community/CommunityEditorProfile/CommunityEditorProfile.js

```diff
@@ -0,0 +1,123 @@
+import React, { useEffect, useState } from 'react';
+import { useForm, useWatch } from 'react-hook-form';
+import { ActionButton } from 'components';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { ProfileSchema, initialValues } from './FormConfig';
+import ProfileForm from './ProfileForm';
+
+// map to match server fields for updating
+const ServerFieldsMap = {
+  communityName: 'name',
+  communityDescription: 'body',
+  communityCategory: 'category',
+  communityTerms: 'termsAndConditionsUrl',
+  banner: 'bannerImgUrl',
+  logo: 'logo',
+};
+
+export default function CommunityEditorProfile({
+  name,
+  body = '',
+  logo,
+  banner,
+  category,
+  terms,
+  // fn to update community payload
+  updateCommunity,
+  // fn to upload image
+  uploadFile,
+} = {}) {
+  const [isUpdatingImage, setIsUpdatingImage] = useState(false);
+  const [isUpdatingBanner, setIsUpdatingBanner] = useState(false);
+
+  const { register, handleSubmit, formState, control, setValue, reset } =
+    useForm({
+      defaultValues: {
+        ...initialValues,
+        communityName: name,
+        communityDescription: body,
+        communityCategory: category,
+        logo: logo ? { imageUrl: logo } : undefined,
+        banner: banner ? { imageUrl: banner } : undefined,
+        communityTerms: terms,
+      },
+      resolver: yupResolver(ProfileSchema),
+    });
+
+  const logoField = useWatch({ control, name: 'logo' });
+  const bannerField = useWatch({ control, name: 'banner' });
```

Comment:
image file has been added to form so we don't need a state to save it. We need to use useWatch to get updates and trigger renders if a new image is uploaded since we can't use register on input used by react-drop-zone